h/views/accounts.py

```diff
@@ -139,15 +137,24 @@ def logout(self):
 
     def _redirect_if_logged_in(self):
         if self.request.authenticated_userid is not None:
-            raise httpexceptions.HTTPFound(location=self.login_redirect)
+            raise httpexceptions.HTTPFound(location=self._login_redirect())
+
+    def _login_redirect(self):
+        return self.request.params.get('next', _login_redirect_url(self.request))
```

Comment:
According to the linked Github issue, this should be `/users/:username/search`, but it is currently `/search`